invenio_records_rest/serializers/datacite.py

```diff
@@ -75,15 +80,42 @@ def serialize_oaipmh(self, pid, record):
             if isinstance(record['_source'], Record) \
             else self.transform_search_hit(pid, record)
 
-        return schema31.dump_etree(obj)
+        return self.schema.dump_etree(obj)
+
+
+class DataCite31Serializer(BaseDataCiteSerializer):
+    """Marshmallow DataCite serializer v3.1 for records.
+
+    Note: This serializer is not suitable for serializing large number of
+    records.
+    """
+
+    schema = schema31
+    """Class variable to define which schema use."""
+    version = '3.1'
+    """Class variable to tell the version of the schema."""
+
+
+class DataCite40Serializer(BaseDataCiteSerializer):
+    """Marshmallow DataCite serializer v4.0 for records.
+
+    Note: This serializer is not suitable for serializing large number of
+    records.
+    """
+
+    schema = schema40
+    """Class variable to define which schema use."""
+    version = '4.0'
+    """Class variable to tell the version of the schema."""
 
 
 class OAIDataCiteSerializer(object):
     """OAI DataCite serializer (only for OAI-PMH)."""
 
-    def __init__(self, v31=None, datacentre=None, is_reference_quality='true'):
+    def __init__(self, serializer=None, datacentre=None,
```

Comment:
Each instance has to provide their own DataCite serializer (which will be either v31 or v4). See e.g. current [usage example](https://github.com/zenodo/zenodo/blob/master/zenodo/modules/records/serializers/__init__.py#L77)